packages/sdk/src/sync-agent/timeline/models/timeline-types.ts

```diff
@@ -0,0 +1,396 @@
+    SpaceImage = 'm.space.image',
+    SpaceParent = 'm.space.parent',
+    SpaceUsername = 'm.space.username',
+    SpaceDisplayName = 'm.space.display_name',
+    SpaceEnsAddress = 'm.space.ens_name',
+    SpaceNft = 'm.space.nft',
+    Unpin = 'm.unpin',
+}
+
+export interface MiniblockHeaderEvent {
+    kind: RiverEvent.MiniblockHeader
+    message: MiniblockHeader
+}
+
+export interface FulfillmentEvent {
+    kind: RiverEvent.Fulfillment
+    sessionIds: string[]
+    deviceKey: string
+    to: string
+    from: string
+}
+
+export interface KeySolicitationEvent {
+    kind: RiverEvent.KeySolicitation
+    sessionIds: string[]
+    deviceKey: string
+    isNewDevice: boolean
+}
+
+export interface RoomCreateEvent {
+    kind: RiverEvent.RoomCreate
+    creatorId: string
+    type?: PayloadCaseType
+    spaceId?: string // valid on casablanca channel streams
+}
+
+export interface ChannelCreateEvent {
+    kind: RiverEvent.ChannelCreate
+    creatorId: string
+    channelId: string
+    channelOp?: ChannelOp
+    channelSettings?: SpacePayload_ChannelSettings
+}
+export interface ReactionEvent {
+    kind: RiverEvent.Reaction
+    targetEventId: string
+    reaction: string
+}
+
+export interface SpaceUsernameEvent {
+    kind: RiverEvent.SpaceUsername
+    userId: string
+    username: string
+}
+
+export interface SpaceDisplayNameEvent {
+    kind: RiverEvent.SpaceDisplayName
+    userId: string
+    displayName: string
+}
+
+export interface SpaceEnsAddressEvent {
+    kind: RiverEvent.SpaceEnsAddress
+    userId: string
+    ensAddress: Uint8Array
+}
+
+export interface SpaceNftEvent {
+    kind: RiverEvent.SpaceNft
+    userId: string
+    contractAddress: string
+    tokenId: string
+    chainId: number
+}
+
+export interface PinEvent {
+    kind: RiverEvent.Pin
+    userId: string
+    pinnedEventId: string
+}
+
+export interface UnpinEvent {
+    kind: RiverEvent.Unpin
+    userId: string
+    unpinnedEventId: string
+}
+
+export interface RoomMessageEncryptedEvent {
+    kind: RiverEvent.RoomMessageEncrypted
+    error?: DecryptionSessionError
+}
+
+export interface RoomMessageEncryptedRefEvent {
+    kind: RiverEvent.RoomMessageEncryptedWithRef
+    refEventId: string
+}
+
+export interface RoomPropertiesEvent {
+    kind: RiverEvent.RoomProperties
+    properties: ChannelProperties
+}
+
+// TODO: membership here doenst map 1-1 to MembershipOp
+export enum Membership {
+    Join = 'join',
+    Invite = 'invite',
+    Leave = 'leave',
+    Ban = 'ban',
+    // NOTE: 'Knock' got removed, isnt used anywhere
+    None = '',
+}
+
+export interface RoomMemberEvent {
+    kind: RiverEvent.RoomMember
+    userId: string
+    initiatorId: string
+    membership: Membership
+    streamId?: string // in a case of an invitation to a channel with a streamId
+}
+
+// TODO: understand, what is Message.GM?
+// TODO: better naming maybe?
+export enum MessageType {
+    Text = 'm.text',
+    // GM = 'm.gm',
+    Image = 'm.Image',
+}
```

Comment:
What Message.GM is?

Reply:
it's a protobuf "any" meant to allow extending the protocol with whatever you want to send.